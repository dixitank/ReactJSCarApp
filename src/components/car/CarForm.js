import React, { PropTypes } from 'react';
import { Field, reduxForm } from 'redux-form';
import FieldInput from '../common/FieldInput';

export const CarForm = ({ handleSubmit, pristine, reset, submitting, heading, handleSave, handleCancel }) => {
    return (
        <form onSubmit={handleSubmit(handleSave)}>
            <h1>{heading}</h1>

            <Field
                type="text"
                name="model"
                label="Car model"
                placeholder="Car Model"
                component={FieldInput}
            />

            <Field
                type="text"
                name="brand"
                label="Car brand"
                placeholder="Car brand"
                component={FieldInput}
            />

            <Field
                type="text"
                name="version"
                label="Car version"
                placeholder="Version of Car"
                component={FieldInput}
            />
            <Field
                type="number"
                name="year"
                label="Year of release"
                placeholder="Year of release"
                component={FieldInput}
            />

            <Field
                type="number"
                name="price"
                label="Price"
                placeholder="Price"
                component={FieldInput}
            />
            <Field
                type="number"
                name="fuel"
                label="Fuel Consumption"
                placeholder="Fuel Consumption in miles/Litre"
                component={FieldInput}
            />
            <Field
                type="number"
                name="annualMaintainence"
                label="Annual maintainence"
                placeholder="Annual Maintainence in $"
                component={FieldInput}
            />

            <div>
                <button type="submit" disabled={submitting} className="btn btn-primary"><i className="fa fa-paper-plane-o" aria-hidden="true" /> Submit</button>

                {heading === 'Add' && <button type="button" disabled={pristine || submitting} onClick={reset} className="btn btn-default btn-space">Clear Values</button>}

                <button type="button" className="btn btn-default btn-space" onClick={handleCancel}>Cancel</button>
            </div>
        </form>
    );
};





const validate = values => {
    const errors = {};

    if (!values.model) {
        errors.model = 'Required';
    }

    if (!values.make) {
        errors.make = 'Required';
    }

    if (!values.version) {
        errors.version = 'Required';
    }

    if (!values.year) {
        errors.year = 'Required';
    }

    if (!values.price) {
        errors.price = 'Required';
    }

    if (!values.fuel) {
        errors.fuel = 'Required';
    }

    if (!values.annualMaintainence) {
        errors.annualMaintainence = 'Required';
    }

    return errors;
};



CarForm.propTypes = {
    handleSubmit: PropTypes.func.isRequired,
    pristine: PropTypes.bool.isRequired,
    reset: PropTypes.func.isRequired,
    submitting: PropTypes.bool.isRequired,
    heading: PropTypes.string.isRequired,
    handleSave: PropTypes.func.isRequired,
    handleCancel: PropTypes.func.isRequired
};



export default reduxForm({
    form: 'CarForm',
    validate
})(CarForm);
