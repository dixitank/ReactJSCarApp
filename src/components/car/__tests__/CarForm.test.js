import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {CarForm} from '../CarForm';



function setup(heading) {
    const props = {
        handleSubmit: jest.fn(),
        pristine: true,
        reset: jest.fn(),
        submitting: false,
        heading: heading,
        handleSave: jest.fn(),
        handleCancel: jest.fn(),
    };

    return shallow(<CarForm {...props}/>);
}


describe('CarForm.test.js', () => {

    it('renders without crashing', () => {
        const wrapper = setup('Add');
        expect(wrapper.length).toEqual(1);
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });


    it('renders form and display "Add" in h1', () => {
        const wrapper = setup('Add');
        expect(wrapper.find('h1').text()).toEqual('Add');
    });
    
});

