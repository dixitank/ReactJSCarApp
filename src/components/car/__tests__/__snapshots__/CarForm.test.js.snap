exports[`CarForm.test.js renders without crashing 1`] = `
<form>
  <h1>
    Add
  </h1>
  <Field
    component={[Function]}
    label="Car model"
    name="model"
    placeholder="Car Model"
    type="text"
    validate={
      Array [
        [Function],
        [Function],
      ]
    } />
  <Field
    component={[Function]}
    label="Car brand"
    name="brand"
    placeholder="Car brand"
    type="text"
    validate={
      Array [
        [Function],
        [Function],
      ]
    } />
  <Field
    component={[Function]}
    label="Car version"
    name="version"
    placeholder="Version of Car"
    type="text"
    validate={
      Array [
        [Function],
      ]
    } />
  <Field
    component={[Function]}
    label="Year of release"
    name="year"
    placeholder="Year of release"
    type="number"
    validate={
      Array [
        [Function],
      ]
    } />
  <Field
    component={[Function]}
    label="Price"
    name="price"
    placeholder="Price"
    type="number"
    validate={
      Array [
        [Function],
      ]
    } />
  <Field
    component={[Function]}
    label="Fuel Consumption"
    name="fuel"
    placeholder="Fuel Consumption in miles/Litre"
    type="number"
    validate={
      Array [
        [Function],
      ]
    } />
  <Field
    component={[Function]}
    label="Annual maintainence"
    name="annualMaintainence"
    placeholder="Annual Maintainence in $"
    type="number"
    validate={
      Array [
        [Function],
      ]
    } />
  <div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit">
      <i
        aria-hidden="true"
        className="fa fa-paper-plane-o" />
       Submit
    </button>
    <button
      className="btn btn-default btn-space"
      disabled={true}
      onClick={[Function]}
      type="button">
      Clear Values
    </button>
    <button
      className="btn btn-default btn-space"
      onClick={[Function]}
      type="button">
      Cancel
    </button>
  </div>
</form>
`;
